### PR TITLE
Rename StoragePolicies to StorageClasses in Content Library

### PR DESCRIPTION
**Pull request by adambarreiro**
The main goal of this PR is two address two problems:
- Field StoragePolicies has been renamed to StorageClasses in ContentLibrary type. This PR fixes the errors that this change causes in the code as it was.
- RegionStoragePolicy now is used within a `Region` (is a method of `Region` type). This is because a given Region Storage Policy belongs to a Region, as it can be a Region Storage Policy with same name in different regions, causing problems.

However, a some side effects have been detected:

- A new error when creating and deleting Content Libraries. The existing workaround has been updated.
- A new error when fetching a non-existing Content Library. This one is ignored in tests.
- The renaming of StoragePolicy to StorageClass causes some inconsistencies reflected in a new TODO in the Content library test.
- A new glitch in VCDClient.GetAllContentLibraries API endpoint returns always `IsShared: false`

I also refactored existing function `getOrCreateVcAndNsxtManager` into two individual functions with a single responsibility: `getOrCreateVCenter` and `getOrCreateNsxtManager`, and added a new one `getOrCreateRegion`. This could eventually help to introduce generics to avoid repeating code. Also, these functions now return the cleanup function itself, so cleaning up afterwards is trivial, one only needs to `defer cleanup()` where _cleanup_ is the returned function.

Tests with `make testtm` passed.

### CODE REVIEW

File: govcd/tm_content_library_test.go [hunk 1]
```diff
@@ -74,8 +88,59 @@ func (vcd *TestVCD) Test_ContentLibraryProvider(check *C) {
...
+	cl, err = vcd.client.GetContentLibraryById(cl.ContentLibrary.ID)
 	check.Assert(err, IsNil)
 	check.Assert(cl, NotNil)
 	check.Assert(*cl.ContentLibrary, DeepEquals, *createdCl.ContentLibrary)
 }
+
+// getOrCreateRegion will check configuration file and create a Region if
+// stated in the 'createRegion' testing property not present in TM.
+// Otherwise, it just retrieves it
+func getOrCreateRegion(vcd *TestVCD, nsxtManager *NsxtManagerOpenApi, supervisor *Supervisor, check *C) (*Region, func()) {
```
[the remark this answers is not in the document]
Comment (adambarreiro) [reply govcd/tm_content_library_test.go:100]: Yeah, I agree with moving. Regarding combining "get or create", it's the main point of the function: we need a Region (or any other item) but we don't care if it exists or it was created. Or in other words, we need to abstract where the Region (or other item) comes from.

[the remark this answers is not in the document]
Comment (Didainius) [reply govcd/tm_content_library_test.go:100]: Yes, @lvirbalas  `getOrCreateRegion` is intentionally such method that should possibly avoid building pyramids.

I am ok with moving to separate files. Could be something like `tm_test.go` or so

**Review by Didainius — APPROVED** [review]

File: govcd/tm_content_library_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (adambarreiro) [reply govcd/tm_content_library_test.go]: Moved to `tm_common_test.go`

[the remark this answers is not in the document]
Comment (lvirbalas) [reply govcd/tm_content_library_test.go]: Awesome, thanks!

**Review by lvirbalas — APPROVED** [review]

**Pull request merged**